### PR TITLE
Update Angular Version from v17 to v18

### LINKED ISSUE #226: Upgrade angular version to v18
**Description**
[A clear and concise description of what you're missing.](https://angular.dev/update-guide?v=17.0-18.0&l=3)

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

_If this issue requests multiple changes consider using checkboxes or lists._

### PR DESCRIPTION
**Pull request by lpandath**
## Description

This PR updates our Angular application from version 17 to 18.

#### What does this PR do?

Upgrading the Angular core and CLI to version 18 using ng update @angular/core@18 @angular/cli@18.
Updating TypeScript to version 5.4 or newer.
Replacing the deprecated async from @angular/core with waitForAsync.
Attempting integration with M3 Material UI, though further features are necessary for global configurations.

#### Code review focus

Version compatibility
Pay attention to the removal of async and the adoption of waitForAsync
The integration of M3 Material UI is partial and needs review for complete and coherent implementation across the application.

#### Dependencies

npm run ci:install emits some warnings that could be linked to version mismatches or deprecated features. 

Material UI Compatibility: The M3 Material UI requires extensive configuration changes which are not covered fully in this PR and will be addressed separately.

### Checks

- [ ] Summary updated
- [ ] Version view updated
- [ ] Documentation added
- [ ] Tests added
- [ ] E2e tests created
- [x] Successfully ran e2e tests before merge

closes GH-

### CODE REVIEW

File: apps/damap-frontend/src/app/components/layout/layout.component.css [hunk 1]
```diff
@@ -35,7 +35,7 @@ mat-nav-list {
 }
 
 .mm-navbars-top {
-  background-color: var(--main-color);
+  background-color: #006699;
```
Comment (rekt-hard) [inline apps/damap-frontend/src/app/components/layout/layout.component.css]: Is this required for the update? Might lose customization possibility here (same for the change further down the file)

**Review by rekt-hard — COMMENTED** [review]
good job! just a minor comment.

when installing the dependencies, there are some warnings about deprecated libraries. Would it make sense to run `npm audit fix` to update them?

File: apps/damap-frontend/src/app/components/layout/layout.component.css (hunk 1 above)
Comment (lpandath) [reply apps/damap-frontend/src/app/components/layout/layout.component.css]: @rekt-hard about `audit fix` i did that but somehow it showed me again the warning. Ill push the changes. 

About the color I'll revert that, was just for testing while trying M3.

**Review by rekt-hard — DISMISSED** [review]
good job. will squash and merge

**Review by rekt-hard — COMMENTED** [review]
Ah one last thing: The Dockerfile should also be updated to use an Angular 18 image and the same nx version we use in the project.

https://github.com/tuwien-csd/damap-frontend/blob/db1a413aafb6d9d1980c6d7222cf37d05a36a3e6/Dockerfile#L5

https://github.com/tuwien-csd/damap-frontend/blob/db1a413aafb6d9d1980c6d7222cf37d05a36a3e6/Dockerfile#L26

**Comment by lpandath** [comment]
> Ah one last thing: The Dockerfile should also be updated to use an Angular 18 image and the same nx version we use in the project.
> 
> https://github.com/tuwien-csd/damap-frontend/blob/db1a413aafb6d9d1980c6d7222cf37d05a36a3e6/Dockerfile#L5
> 
> https://github.com/tuwien-csd/damap-frontend/blob/db1a413aafb6d9d1980c6d7222cf37d05a36a3e6/Dockerfile#L26

Oh yes! Ill change this.

**Review by rekt-hard — APPROVED** [review]

**Pull request merged**